app/dashboard/sync/casper.py

```diff
@@ -0,0 +1,63 @@
+import logging
+
+from django.utils import timezone
+
+import requests
+from dashboard.sync.helpers import record_payout_activity
+
+logger = logging.getLogger(__name__)
+
+
+def get_casper_txn_status(fulfillment):
+    txnid = fulfillment.payout_tx_id
+    token_name = fulfillment.token_name
+    funderAddress = fulfillment.funder_address
+    amount = fulfillment.payout_amount
+    payeeAddress = fulfillment.fulfiller_address
+
+    if token_name != 'CSPR' or not txnid:
+        return None
+
+    try:
+        data = {
+            'id': 0,
+            'jsonrpc': '2.0',
+            'method': 'info_get_deploy',
+            'params': [ txnid ]
+        }
+
+        casper_response = requests.post('http://3.142.224.108:7777/rpc', json=data).json()
+
+        result = casper_response['result']
+
+        if result:
+            if (
+                result["deploy"]["hash"] == txnid
+                and result["deploy"]["header"]["account"] == funderAddress
```

Comment:
Do we wanna check if the values exist ? Else we might end up errors.
Asking cause I haven't used the rpc endpoint -> so not sure if those would always be present

Reply:
according to the docs, yes.

i used get for the optional key "Success" -> `if result["execution_results"][0]["result"].get("Success", False) != False:`